### ISSUE TITLE
Is there a way for Knomi to show temp by Tool?

### ISSUE THREAD

**Issue by TheSin-** [bigtreetech/KNOMI#16]
I'd like to add a knot to each of my tools and show the temp of the tool it's attached to, is there a way to accomplish this easily?  To T0 will show the temp for T0 and T1 will show the temp for T1 etc etc

**Comment by M4dMike** [comment]
Sounds like a cool idea :)

This would require costume code on each Knomi, but can definitely be achieved.

Have a look at line 1007 to 1015 in [main.cpp](https://github.com/bigtreetech/KNOMI/blob/master/Firmware/src/main.cpp)

    String nameStr3 = doc["temperature"]["tool0"]["actual"].as<String>();
    String nameStr4 = doc["temperature"]["tool0"]["target"].as<String>();
    String nameStr5 = doc["state"]["flags"]["printing"].as<String>();
    String nameStr6 = doc["state"]["flags"]["paused"].as<String>();

    bedtemp_actual = (uint16_t)((doc["temperature"]["bed"]["actual"].as<double>())*100);
    bedtemp_target = (uint16_t)((doc["temperature"]["bed"]["target"].as<double>())*100);
    tooltemp_actual = (uint16_t)((doc["temperature"]["tool0"]["actual"].as<double>())*100);
    tooltemp_target = (uint16_t)((doc["temperature"]["tool0"]["target"].as<double>())*100);`

As you can see, its referencing temps for `tool0` -> change that too `tool1` and build + flash it to your second KNOMI

**Comment by TheSin-** [comment]
excellent!  I'll order up an other one to see if I can figure it out, this will be great if I can get it running!

**Comment by TheSin-** [comment]
Do you know of an example to make it a configuration option on startup, like the screen used for the wifi setup?  That way they can all have one firmware that I can just select the tool id on?

Trying to figure out the codebase for configuration options.

**Comment by M4dMike** [comment]
Exactly, adding a 4th input field and storing this information makes it dynamic and would probably help to get this change back into mainline. Awesome!

**Comment by TheSin-** [comment]
Into the testing phase now!  Thanks again for leading me into the right direction, I'm going to make a PR once I know 100% it's working and saving.

![IMG_8783](https://github.com/bigtreetech/KNOMI/assets/1868661/917286ee-3d16-4989-a773-5771002cfc6e)

**Comment by M4dMike** [comment]
Awesome 😀
Happy to help, knomi has a lot of unused potential 😉

**Comment by mcbeenb** [comment]
Awesome work guys!!
